packages/nango-yaml/lib/helpers.ts

```diff
@@ -0,0 +1,192 @@
+import path from 'node:path';
+import ms from 'ms';
+import type { StringValue } from 'ms';
+import type { NangoYaml, NangoYamlParsed, NangoYamlParsedIntegration } from '@nangohq/types';
+
+interface IntervalResponse {
+    interval: StringValue;
+    offset: number;
+}
+
+export function determineVersion(configData: NangoYaml): 'v1' | 'v2' {
+    if (!configData.integrations) {
+        return 'v1';
+    }
+
+    const keys = Object.keys(configData.integrations);
+    if (keys.length <= 0) {
+        // Actually not possible to know
+        return 'v1';
+    }
+
+    const firstProviderConfig = configData.integrations[keys[0]!];
+
+    if (firstProviderConfig && ('syncs' in firstProviderConfig || 'actions' in firstProviderConfig)) {
+        return 'v2';
+    } else {
+        return 'v1';
+    }
+}
+
+export function getOffset(interval: StringValue, date: Date): number {
+    const intervalMilliseconds = ms(interval);
+
+    const nowMilliseconds = date.getMinutes() * 60 * 1000 + date.getSeconds() * 1000 + date.getMilliseconds();
+
+    const offset = nowMilliseconds % intervalMilliseconds;
+
+    if (isNaN(offset)) {
+        return 0;
+    }
+
+    return offset;
+}
+
+/**
+ * Get Interval
+ * @desc get the interval based on the runs property in the yaml. The offset
+ * should be the amount of time that the interval should be offset by.
+ * If the time is 1536 and the interval is 30m then the next time the sync should run is 1606
+ * and then 1636 etc. The offset should be based on the interval and should never be
+ * greater than the interval
+ */
+export function getInterval(runs: string, date: Date): IntervalResponse | Error {
```

Comment:
Moved from shared, with the diff that ServiceResponse does not exists and not available in CLI so I just return an Error if any